src/set.cr

```diff
@@ -136,17 +136,17 @@ struct Set(T)
     @hash.has_key?(object)
   end
 
-  # Removes the *object* from the set and returns `self`.
+  # Deletes the *object* and returns it, otherwise returns nil.
```

Comment:
```suggestion
  # Removes the *object* from the set - if found - and returns it, otherwise returns `nil`.
```

Reply:
I don't think is is necessary to add `if found`. It is logical to only remove an object if it exists, the opposite is not possible. `Hash#delete` doesn't have it.
Otherwise, valid suggestion.

Reply:
Hehe. I know this is safe. I knew it wasn't safe (because I implemented the compiler!) to add type restrictions in a block argument. So they are different things.

Reply:
You removed the part saying where you remove this object from.